File: 2299-merge-nodes-in-between-zeros/2299-merge-nodes-in-between-zeros.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeNodes(ListNode head) {
        ListNode t = head.next;
        ListNode ans = head;
        int sum;
        while (t != null) {
            sum = 0;
            while (t.val != 0) {
                sum += t.val;
                t = t.next;
            }
            t = t.next;
            head.val = sum;
            if (t == null) head.next = null;
            else head = head.next;
        }
        return ans;
    }
}

File: 2299-merge-nodes-in-between-zeros/2299-merge-nodes-in-between-zeros.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* mergeNodes(struct ListNode* head) {
    if (!head) return head;
    struct ListNode* t = head->next;
    struct ListNode* ans = head;
    int sum;
    while (t) {
        sum = 0;
        while (t->val) {
            sum += t->val;
            t = t->next;
        }
        t = t->next;
        head->val = sum;
        if (!t) head->next = NULL;
        else head = head->next;
    }
    return ans;
}